Examples/TicTacToe/Sources/Core/LoginCore.swift

```diff
@@ -94,7 +94,7 @@ public let loginReducer =
 
       case .twoFactorDismissed:
         state.twoFactor = nil
-        return .none
+        return .cancel(id: TwoFactorTearDownToken())
```

Comment:
This approach works well for this particular case (that of a single long lived effect in an immediate child), but what about an alternative where you send an action to the child before nilling it out? Pseudocode:
```
case childDismissed:
send teardown action to child
then nil out state.child
```

This seems like it may be easier to extract to a higher order reducer (like the one that handles lifecycles). That also serves to encapsulate the teardown logic in the child. Also, if you had a deeply nested hierarchy it would allow the parent to tell it's child to teardown, which would in turn tell it's own child to teardown etc.